### PR TITLE
Rush validation

### PR DESCRIPTION
**Pull request by yousiefc**
# Description

This PR includes the following proposed change(s):

- Rush now redirects correctly
- Rush data now submitted on continue
- Prepopulation issue resolved

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [x] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/frontend/efiling-frontend/src/components/page/rush/Rush.js [hunk 1]
```diff
@@ -105,7 +106,45 @@ export default function Rush({ payment }) {
...
+      formData.append("files", files[i]);
+    }
+    const req = {
+      rushType: fields.rushType,
+      firstName: fields.firstName,
+      lastName: fields.surname,
+      organization: fields.org,
+      phoneNumber: fields.phoneNumber,
+      email: fields.email,
+      country: fields.country,
```
Comment (BrendanBeachBC) [inline src/frontend/efiling-frontend/src/components/page/rush/Rush.js:121]: During the phone number validation fix, I switched country from being a string to being an object with code and description fields. Does this need to be switched to be `fields.country.description` or does the rush submission work with the object?

**Review by BrendanBeachBC — DISMISSED** [review]
Looks good! One question related to a change that I made to the rush page recently

File: src/frontend/efiling-frontend/src/components/page/rush/Rush.js [hunk 2]
```diff
@@ -511,7 +546,9 @@ export default function Rush({ payment }) {
           label="The attached application is made under Rule 8-5 (1) SCR."
           onSelect={() => {
             setRadioStatusComponents();
+            setFields({ ...fields, rushType: "rule" });
             setRadio1(true);
+            resetFields();
```
Comment (TayGov) [inline src/frontend/efiling-frontend/src/components/page/rush/Rush.js:551]: This should be re-ordered from:
setFields({ ...fields, rushType: "rule" });
setRadio1(true);
resetFields();

To:
resetFields();
setFields({ ...fields, rushType: "rule" });
setRadio1(true);

The reset is actually clearing the rushType. This applies to all three radio buttons.

**Review by TayGov — CHANGES_REQUESTED** [review]
Just the one thing about the radio buttons

**Review by TayGov — APPROVED** [review]
👍

**Pull request merged**